Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -0,0 +1,240 @@
+    ///
+    /// - Returns: `ViewAnnotationOptions` if view was found and `nil` otherwise.
+    public func options(byAnnotationView view: AnnotationView) -> ViewAnnotationOptions? {
+        return annotationIdByViews[view].flatMap { try? mapboxMap.options(forViewAnnotationWithId: $0) }
+    }
+
+    // MARK: - Internal functions
+
+    internal func placeAnnotations(positions: [ViewAnnotationPositionDescriptor]) {
+        // Iterate through and update all view annotations
+        // First update the position of the views based on the placement info from GL-Native
+        // Then hide the views which are off screen
+        var visibleAnnotationIds: Set<String> = []
+
+        for position in positions {
+            validateAnnotation(byAnnotationId: position.identifier)
+            guard let annotationView = annotationViewsById[position.identifier] else {
+                continue
+            }
+            annotationView.translatesAutoresizingMaskIntoConstraints = true
+            annotationView.frame = CGRect(
+                origin: position.leftTopCoordinate.point,
+                size: CGSize(width: CGFloat(position.width), height: CGFloat(position.height))
+            )
+            annotationView.setInternalVisibility(isHidden: false)
+            visibleAnnotationIds.insert(position.identifier)
+        }
+
+        let annotationsToHide = Set<String>(annotationViewsById.keys).subtracting(visibleAnnotationIds)
+        for id in annotationsToHide {
+            validateAnnotation(byAnnotationId: id)
+            if let annotationView = annotationViewsById[id] {
+                annotationView.setInternalVisibility(isHidden: true)
+            }
+        }
+    }
+
+    internal func validateAnnotation(byAnnotationId id: String) {
+        guard let annotationView = annotationViewsById[id] else { return }
+        // If the user explicitly removed the ViewAnnotation or it's wrapped view
+        // we need to remove it from our layout calculation
+        if annotationView.subviews.isEmpty || annotationView.superview == nil {
+            try? remove(annotationView)
+        }
+        // View is still considered for layout calculation, users should not modify the visibility of the wrapped view
+        if let wrappedView = annotationView.subviews.first, wrappedView.isHidden {
+            Log.warning(forMessage: "Visibility changed for wrapped view", category: "Annotations")
+        }
+    }
+
+}
+
+extension ViewAnnotationManager: DelegatingViewAnnotationPositionsUpdateListenerDelegate {
+
+    internal func onViewAnnotationPositionsUpdate(forPositions positions: [ViewAnnotationPositionDescriptor]) {
+        placeAnnotations(positions: positions)
+    }
+
+}
+
+public final class AnnotationView: SubviewInteractionOnlyView {
+
+    private static var currentId = 0
+    internal let id: String = {
+        let id = String(currentId)
+        currentId += 1
+        return id
+    }()
```

Comment:
Let's try to avoid using static storage. A simple solution would be for ViewAnnotationManager to have an instance property to keep track of `currentId` and pass that into AnnotationView's init.

Reply:
Good point I'll change it. But also we have to make sure we don't use the same ID twice, because gl-native uses them to uniquely identify the managed view annotations. Since we will only have one ViewAnnotationManager per map, I think it's fine to just reset it now. But I'll add a comment about it.

Reply:
Actually, looking at this more, it seems unnecessary to store the id on the AnnotationView at all